server/src/main/java/org/opensearch/action/search/TransportMultiSearchAction.java

```diff
@@ -193,7 +195,7 @@ private void handleResponse(final int responseSlot, final MultiSearchResponse.It
                 if (responseCounter.decrementAndGet() == 0) {
                     assert requests.isEmpty();
                     finish();
-                } else {
+                } else if (isCancelled(request.request.getParentTask()) == false) {
```

Comment:
I think this makes sense but we need to populate the response slot with the error (task was cancelled) or alike, what do you think?

Reply:
I could be mistaken, but I think if the parent task gets canceled, then tan error response is already sent back to the client.

The customer ticket that prompted this issue involved the client closing the connection (due to timeout).

I'll try hacking something together with explicit task cancelation to see what the client receives.

Reply:
I am not very familiar with this API, but intuitively, we  may have a case when some requests are completed (before cancellation) and some would be canceled (after cancellation). So we won't be returning partial results, right? (just cancellation for the whole request)

Reply:
You were correct!

I debugged it locally (using a breakpoint in the `handleResponse` method to pause processing between requests). When I canceled the request, no response was returned and the msearching client just waited.

I've updated it to drain the rest of the queue and output a `TaskCancelledException` as the response for each remaining request.